packages/headless/dist/parsed_doc.json

```diff
@@ -0,0 +1,28 @@
+{
+  "controllers": [ "Hidden for this PR" ],
+  "actions": [
+    {
+      "name": "SearchActions",
+      "members": [
+        {
+          "actionKind": "asyncThunk",
+          "name": "executeSearch",
+          "desc": "Executes a search query.",
+          "parameter": {
+            "name": "analyticsAction",
+            "type": "SearchAction",
+            "desc": "The analytics action to log after a successful query.",
+            "isOptional": false,
+            "kind": "primitive"
+          }
+        },
```

Comment:
Two questions about the output:
1. Do we need the `actionKind`? Focusing again the usage perspective, will a user of Headless ever need to know that?
2. What about the parameter type? My impression is that `SearchAction` is some subset of all actions, so it's important to indicate it. However, without documentation on what a `SearchAction` is, it doesn't mean anything right now (i.e. a user would see that but not know which actions belong to that category). We need to come up with a strategy for how this could be indicated...

Reply:
> Do we need the `actionKind`? Focusing again the usage perspective, will a user of Headless ever need to know that?

The actionKind tells whether an action is asynchronous (AsyncThunk) or a normal action (Action?). I guess when dispatching the action, it's not relevant to know (unless we intend to let people `await` actions). I could remove it.

> What about the parameter type? My impression is that `SearchAction` is some subset of all actions, so it's important to indicate it. However, without documentation on what a `SearchAction` is, it doesn't mean anything right now (i.e. a user would see that but not know which actions belong to that category). We need to come up with a strategy for how this could be indicated...

I haven't added logic to extract interfaces yet, I forgot to mention in the description of this PR that it's not intended to be merged yet.

However, even if I added that logic, the type of `SearchAction` is very complex and documenting its members would be useless. We do need to come up with a strategy on how to document it, it's essentially to return type of certain `log*` actions.

What should we do, in your opinion? Perhaps we could also include a short usage example for `executeSearch` specifically in its description.

Reply:
I don't think it matters what the underlying type of `SearchAction` is. What seems to matter is which other actions are type `SearchAction`, right? So if we had some way to indicate which actions are a `SearchAction`, then we wouldn't really need to document what that type is.

_How_ we indicate this depends on the scope. Are all actions that would be labeled as a `SearchAction` under one namespace, or are they scattered? If an action isn't a `SearchAction`, what is it? Are there a lot of actions that take other actions as parameters, or is `executeSearch` the exception?

As a side note, it's confusing to have a namespace called `SearchActions` and an action type called `SearchAction`. My first thought was that `SearchAction` type actions were just all the actions in the `SearchActions` namespace, which I don't think is the case.

Reply:
`SearchAction` are scattered everywhere. I guess part 3 of this story could be to document analytics actions and give them a return type such as `SearchAction`.

As for the confusing name, I already added it as a topic of discussion for the next major release.

Reply:
Hmm... the only time the `SearchAction` matters as a type is when an action is being passed as a parameter to another action, right? Because dispatching each action works the same way?

Maybe we could get away with just having a list somewhere of `SearchAction` that links each listed action to its actual reference documentation... Because then a user looking for a `SearchAction` could go to that list, see all the possibilities at once, but also get the ref docs for each one.

I'm just throwing out ideas because the organization of actions is getting pretty complicated for a concept so simple. We're going to have yet another problem when it comes to actions that take objects as parameters. (Which, by the way, I still see examples of actions that take multiple parameters in the source code, like `updateSearchConfiguration`, so I'm wondering if parameters should not be an array instead)

Try to think about exactly what you'd be looking for as a developer using Headless when you come to this page, and maybe we can set up a meeting to talk about this further.